modules/gitops-argocd-agent-install-principal-component.adoc

```diff
@@ -0,0 +1,326 @@
++
+[source,terminal]
+----
+$ oc apply -f argocd.yaml -n "<PRINCIPAL_NAMESPACE>" --context "<PRINCIPAL_CONTEXT>"
+----
++
+[NOTE]
+====
+At this stage, Argo CD pods start successfully, but the Principal pod remains in a `Running` (not Ready) state because the required secrets have not yet been created. Secrets must be created later because some configurations, such as the principal hostname and resource proxy service names, are available only after the Operator enables the Principal component.
+====
+
+.. Verify Argo CD instance deployment. 
+Check the pods in the Principal namespace by running the following command.
++
+[source,terminal]
+----
+$ oc get pod -n "<PRINCIPAL_NAMESPACE>" --context "<PRINCIPAL_CONTEXT>"
+----
++
+As a result, the following output is displayed:
++
+[source,terminal]
+----
+NAME                                               READY  STATUS 
+openshift-gitops-agent-principal-xxxxxxxxx-xxxxx   0/1    Running
+openshift-gitops-redis-xxxxxxxxx-xxxxx             1/1    Running
+openshift-gitops-repo-server-xxxxxxxxx-xxxxx       1/1    Running
+openshift-gitops-server-xxxxxxxxx-xxxxx            1/1    Running
+----
+
+.. Verify the services created by the operator by running the following command:
++
+[source,terminal]
+----
+$ oc get svc -n "<PRINCIPAL_NAMESPACE>" --context "<PRINCIPAL_CONTEXT>"
+----
++
+As a result, the following output is displayed:
++
+[source,terminal]
+----
+NAME                                            TYPE      PORT(S)
+openshift-gitops-agent-principal                ClusterIP 443/TCP
+openshift-gitops-agent-principal-healthz        ClusterIP 8030/TCP
+openshift-gitops-agent-principal-metrics        ClusterIP 9000/TCP
+openshift-gitops-agent-principal-redisproxy     ClusterIP 6379/TCP
+openshift-gitops-agent-principal-resource-proxy ClusterIP 9090/TCP
+----
+
+.. By default, the Operator creates a route for the Principal, which will be used to access Principal services. Verify the default route for the Principal component by running the following command:
++
+[source,terminal]
+----
+$ oc get route -n "<PRINCIPAL_NAMESPACE>" --context "<PRINCIPAL_CONTEXT>"
+----
++
+As a result, the following output is displayed:
++
+[source,terminal]
+----
+NAME                               HOST/PORT
+openshift-gitops-agent-principal   <HOST_NAME>
+----
+
+.. If you want to disable the route or use a `LoadBalancer` service, update the following section in the Argo CD resource:
++
+[source,yaml]
+----
+kind: ArgoCD
+spec:
+  # (...)
+  argocdAgent:
+    principal:
+      server:
+        service:
+          type: LoadBalancer        
+        route:
+          enabled: false
+----
+
+. Create the required namespaces.
+You must create one namespace for each Agent that you plan to connect to the Principal. The namespace names must match the Agent names. Run the following commands:
++
+[source,terminal]
+----
+$ oc create namespace agent-managed
+$ oc create namespace agent-autonomous
```

Comment:
```suggestion
$ oc create namespace agent-managed --context "<PRINCIPAL_CONTEXT>"
$ oc create namespace agent-autonomous --context "<PRINCIPAL_CONTEXT>"
```

Reply:
Yes, better to provide context instead of assuming that at the time f principal setup default context would be principal only.

Reply:
Fixed.